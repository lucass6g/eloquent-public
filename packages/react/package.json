{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
