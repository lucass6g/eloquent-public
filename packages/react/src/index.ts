export {Switch, Case, Default} from './components/conditional-switch'