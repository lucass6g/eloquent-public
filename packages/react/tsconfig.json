{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
