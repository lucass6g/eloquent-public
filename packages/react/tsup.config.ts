import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})