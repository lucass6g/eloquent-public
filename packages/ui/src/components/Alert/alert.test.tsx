import { render, screen } from "@testing-library/react";
import {it, describe, expect} from "vitest"
import { Alert } from "..";
import { Component } from 'lucide-react'

describe('Alert', () => {
    it('renders default alert', () => {
        const component = render(
            <Alert.Root variant="default">
                <Component data-testid="icon" className="h-4 w-4" />
                <Alert.Title>Atenção!</Alert.Title>
                <Alert.Description>
                    Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
                </Alert.Description>
            </Alert.Root>
        );
        expect(screen.getByText('Atenção!')).toBeInTheDocument();
        expect(screen.getByText('Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.')).toBeInTheDocument();
        expect(screen.getByTestId('icon')).toBeInTheDocument(); 
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders destructive alert', () => {
        const component = render(
            <Alert.Root variant="destructive">
                <Component data-testid="icon" className="h-4 w-4" />
                <Alert.Title>Atenção!</Alert.Title>
                <Alert.Description>
                    Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
                </Alert.Description>
            </Alert.Root>
        );
        expect(screen.getByText('Atenção!')).toBeInTheDocument();
        expect(screen.getByText('Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.')).toBeInTheDocument();
        expect(screen.getByTestId('icon')).toBeInTheDocument(); 
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders alert without title', () => {
        const component  = render(
            <Alert.Root variant="default">
                <Component data-testid="icon" className="h-4 w-4" />
                <Alert.Description>
                    Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
                </Alert.Description>
            </Alert.Root>
        );
        expect(screen.queryByText('Atenção!')).not.toBeInTheDocument();
        expect(screen.getByText('Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.')).toBeInTheDocument();
        expect(screen.getByTestId('icon')).toBeInTheDocument(); 
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders alert without description', () => {
        render(
            <Alert.Root variant="default">
                <Component data-testid="icon" className="h-4 w-4" />
                <Alert.Title>Atenção!</Alert.Title>
            </Alert.Root>
        );
        expect(screen.getByText('Atenção!')).toBeInTheDocument();
        expect(screen.queryByText('Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.')).not.toBeInTheDocument();
        expect(screen.getByTestId('icon')).toBeInTheDocument(); 
    });
    it('renders alert without icon', () => {
        const component = render(
            <Alert.Root variant="default">
                <Alert.Title>Atenção!</Alert.Title>
                <Alert.Description>
                    Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
                </Alert.Description>
            </Alert.Root>
        );
        expect(screen.getByText('Atenção!')).toBeInTheDocument();
        expect(screen.getByText('Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.')).toBeInTheDocument();
        expect(screen.queryByRole('svg')).not.toBeInTheDocument(); 
        expect(component.baseElement).toMatchSnapshot();
    });
});