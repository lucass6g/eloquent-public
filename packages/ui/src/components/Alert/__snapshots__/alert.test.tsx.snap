// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > renders alert without icon 1`] = `
<body>
  <div>
    <div
      class="relative w-full p-4 text-sm rounded-[8px] [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:stroke-foreground [&>svg~*]:pl-7 bg-white border border-[#E0E0E0] text-foreground"
      role="alert"
    >
      <h5
        class="text-base not-italic font-medium"
      >
        Atenção!
      </h5>
      <div
        class="text-base font-normal not-italic text-[#616161]"
      >
        Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert > renders alert without title 1`] = `
<body>
  <div>
    <div
      class="relative w-full p-4 text-sm rounded-[8px] [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:stroke-foreground [&>svg~*]:pl-7 bg-white border border-[#E0E0E0] text-foreground"
      role="alert"
    >
      <svg
        class="lucide lucide-component h-4 w-4"
        data-testid="icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
        />
        <path
          d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
        />
        <path
          d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
        />
        <path
          d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
        />
      </svg>
      <div
        class="text-base font-normal not-italic text-[#616161]"
      >
        Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert > renders default alert 1`] = `
<body>
  <div>
    <div
      class="relative w-full p-4 text-sm rounded-[8px] [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:stroke-foreground [&>svg~*]:pl-7 bg-white border border-[#E0E0E0] text-foreground"
      role="alert"
    >
      <svg
        class="lucide lucide-component h-4 w-4"
        data-testid="icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
        />
        <path
          d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
        />
        <path
          d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
        />
        <path
          d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
        />
      </svg>
      <h5
        class="text-base not-italic font-medium"
      >
        Atenção!
      </h5>
      <div
        class="text-base font-normal not-italic text-[#616161]"
      >
        Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
      </div>
    </div>
  </div>
</body>
`;

exports[`Alert > renders destructive alert 1`] = `
<body>
  <div>
    <div
      class="relative w-full p-4 text-sm rounded-[8px] [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg~*]:pl-7 bg-background border text-destructive border-destructive [&>svg]:stroke-destructive *:text-destructive"
      role="alert"
    >
      <svg
        class="lucide lucide-component h-4 w-4"
        data-testid="icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
        />
        <path
          d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
        />
        <path
          d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
        />
        <path
          d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
        />
      </svg>
      <h5
        class="text-base not-italic font-medium"
      >
        Atenção!
      </h5>
      <div
        class="text-base font-normal not-italic text-[#616161]"
      >
        Houve um problema com o seu cartão de crédito. Por favor, tente novamente mais tarde.
      </div>
    </div>
  </div>
</body>
`;
